src/shared/regions/endpointsProvider.ts

```diff
@@ -0,0 +1,53 @@
+/*!
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as vscode from 'vscode'
+import { getLogger, Logger } from '../logger'
+import { ResourceFetcher } from '../resourcefetcher/resourcefetcher'
+import { Endpoints, loadEndpoints } from './endpoints'
+
+export class EndpointsProvider {
+    private readonly logger: Logger = getLogger()
+    public get onEndpointsUpdated(): vscode.Event<EndpointsProvider> {
+        return this.onEndpointsUpdatedEmitter.event
+    }
+    private readonly onEndpointsUpdatedEmitter: vscode.EventEmitter<EndpointsProvider> = new vscode.EventEmitter()
+    private endpoints: Endpoints | undefined
+
+    /**
+     * @param localFetcher Retrieves endpoints manifest from local sources available to the toolkit. Expected
```

Comment:
Nit: will this get shown in an IDE tooltip? If not, you may want to move it to the top of the class.